### ISSUE TITLE
[BUG] Some gradient is always zero when training keras models.

### ISSUE THREAD

**Issue by AsakusaRinne** [SciSharp/TensorFlow.NET#990]
The minimal example to reproduce it:

```cs
Model model = KerasApi.keras.Sequential(new List<ILayer>()
{
    keras.layers.InputLayer((28, 28, 1)),
    keras.layers.Flatten(),
    keras.layers.Dense(100, "relu"),
    keras.layers.Dense(10),
    keras.layers.Softmax(1)
});

model.summary();

model.compile(new Adam(0.001f), new LossesApi().SparseCategoricalCrossentropy(), new string[] { "accuracy" });

var data_loader = new MnistModelLoader();
var num_epochs = 1;
var batch_size = 50;

var dataset = data_loader.LoadAsync(new ModelLoadSetting
{
    TrainDir = "mnist",
    OneHot = false,
    ValidationSize = 50000,
}).Result;

model.fit(dataset.Train.Data, dataset.Train.Labels, batch_size, num_epochs);
```

Watch the trainable variables of the `model` before and after `model.fit`, the kernel of the first dense layer does not chang at all, while other variables works well with the gradients. The reason behind it is that the gradient of the kernel is always zero during the training process.

Besides, this bug is possibly related with #983
